### PR TITLE
chore: Drop support for python 3.8 on kedro-datasets

### PR DESCRIPTION
**Pull request by DimedS**
## Description
This PR drops support for Python 3.8 on `kedro-datasets`.

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by astrojuanlu** [comment]
I think I know what's happening in RTD: 

https://github.com/kedro-org/kedro-plugins/blob/1f707ff31042e3d6856ee04083959bbab99846cc/kedro-datasets/.readthedocs.yaml#L11

With your permission 🔨

File: kedro-datasets/setup.py [hunk 1]
```diff
@@ -216,14 +215,12 @@ def _collect_requirements(requires):
     "requests~=2.20",
     "ruff~=0.0.290",
     "s3fs>=0.3.0, <0.5",  # Needs to be at least 0.3.0 to make use of `cachable` attribute on S3FileSystem.
-    "snowflake-snowpark-python~=1.0.0; python_version == '3.8'",
```
Comment (astrojuanlu) [inline kedro-datasets/setup.py:219]: This will need to be brought back for the docs to build, for example

```
"snowflake-snowpark-python~=1.0"
```

Comment (DimedS) [reply kedro-datasets/setup.py:219]: yes, thanks for help, @astrojuanlu !

**Comment by astrojuanlu** [comment]
Now this looks like a legitimate issue 🤔 

```
sphinx.errors.SphinxWarning: /home/docs/checkouts/readthedocs.org/user_builds/kedro-datasets/envs/442/lib/python3.9/site-packages/kedro_datasets/partitions/incremental_dataset.py:docstring of kedro_datasets.partitions.incremental_dataset.IncrementalDataset.__init__:1:py:class reference target not found: AbstractDataset
```

**Comment by DimedS** [comment]
> > Now this looks like a legitimate issue 🤔
> > ```
> > sphinx.errors.SphinxWarning: /home/docs/checkouts/readthedocs.org/user_builds/kedro-datasets/envs/442/lib/python3.9/site-packages/kedro_datasets/partitions/incremental_dataset.py:docstring of kedro_datasets.partitions.incremental_dataset.IncrementalDataset.__init__:1:py:class reference target not found: AbstractDataset
> > ```
> 
> I think this might just be a weird case of sphinx not finding class import correctly. On the Kedro side we have a list where we ignore these errors: https://github.com/kedro-org/kedro/blob/main/docs/source/conf.py#L96

thank you, @merelcht , will repeat the same there

**Comment by astrojuanlu** [comment]
Yay! Docs building

**Comment by DimedS** [comment]
> Yay! Docs building

🥳 Thanks to @merelcht !

File: kedro-datasets/RELEASE.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # Upcoming Release
+* Removed support for Python 3.8
 
 ## Major features and improvements
 * Removed support for Python 3.7
```
Comment (merelcht) [inline kedro-datasets/RELEASE.md:5]:
```suggestion

## Major features and improvements
* Removed support for Python 3.7 and 3.8
```

File: kedro-datasets/setup.py [hunk 3]
```diff
@@ -41,8 +41,7 @@ def _collect_requirements(requires):
     "pandas.HDFDataset": [
         PANDAS,
         "tables~=3.6.0; platform_system == 'Windows'",
-        "tables~=3.6, <3.9; platform_system != 'Windows' and python_version<'3.9'",
-        "tables~=3.6; platform_system != 'Windows' and python_version>='3.9'",
+        "tables~=3.6; platform_system != 'Windows'",
```
Comment (merelcht) [inline kedro-datasets/setup.py:44]: Are these two actually different in practice?

**Review by merelcht — APPROVED** [review]
Left two minor comments, but other than that LGTM! 🥳

File: kedro-datasets/setup.py (hunk 3 above)
Comment (astrojuanlu) [reply kedro-datasets/setup.py:44]: Yep, one is capped at <3.7 and the other at <4. However: #426

**Review by astrojuanlu — APPROVED** [review]
Let's get this merged and hopefully look into merging #426 as well

**Pull request merged**